### DISCUSSION TITLE
[@types/semver] Unsafe call of an `any` typed value. (eslint@typescript-eslint/no-unsafe-call)

### DISCUSSION THREAD

**Discussion by DiamondeX** [DefinitelyTyped/DefinitelyTyped#58354] [Issues with a @types package] [answered]
The eslint rule `Unsafe call of an 'any' typed value. (eslint@typescript-eslint/no-unsafe-call)` is not satisfied with the current state of `semver`'s functions type definitions. So this rule is flagged apparently for the usage of all it's functions. At least the code sample below will flag both `lt()` and `clean()` functions:

```ts
import { lt, clean } from 'semver';

describe('Check prerequisites', () => {
  test('node version should be less then 16', () => {

    expect(lt(clean(process.version)||'', '16.0.0')).toStrictEqual(true);
  });
});
```

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
best guess, it looks like your setup is not resolving types properly, so you're getting imported functions resolved to type `any` (hence a linter error). In your editor of choice you should see resolved types:
![image](https://user-images.githubusercontent.com/14539/150601799-d606a6b7-df26-4ec1-9535-5ba4a415edc0.png)

**Comment by DiamondeX** [comment, marked as answer]
After restarting the editor the problem is disappeared. The problem seemed to be in another place.
